### PR TITLE
Replayer UI v3

### PR DESCRIPTION
**Pull request by h3ndrk**
## Introduced Changes

image

Improved code quality, some more features.

## ToDo / Known Issues

Now it's perfect :wink:

## Ideas for Next Iterations (Not This PR)

- Animations

## How to Test

- Record
- Replay
- ...
- Profit

### CODE REVIEW

File: crates/hulk_replayer/src/frames.rs [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+                    input.pointer.button_down(PointerButton::Primary),
+                    input.smooth_scroll_delta,
+                    input.modifiers.shift,
+                    Keys {
+                        jump_backward_large: input.key_pressed(Key::J),
+                        jump_forward_large: input.key_pressed(Key::L),
+                        jump_backward_small: input.key_pressed(Key::ArrowLeft),
+                        jump_forward_small: input.key_pressed(Key::ArrowRight),
+                        step_backward: input.key_pressed(Key::Comma),
+                        step_forward: input.key_pressed(Key::Period),
```
[the remark this answers is not in the document]
Comment (h3ndrk) [reply crates/hulk_replayer/src/frames.rs:213]: Do you mean instead the current ones or in addition? The current ones are inspired from the YouTube video player.

[the remark this answers is not in the document]
Comment (knoellle) [reply crates/hulk_replayer/src/frames.rs:213]: I know, I never found those to be very intuitive.
But if you or or other people want to use it like that, then I guess have up down as additional keys.
Up/down for larger jumps is the way mpv does it.

File: crates/hulk_replayer/src/coordinate_systems.rs [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+};
+
+use derive_more::{Add, AddAssign, Mul, Neg, Rem, Sub, SubAssign};
+use eframe::egui::remap;
+
+/// # Absolute Time Coordinate
+///
+/// Example: Recording Frames
+///
+/// - Offset: `std::time::UNIX_EPOCH``
```
[the remark this answers is not in the document]
Comment (h3ndrk) [reply crates/hulk_replayer/src/coordinate_systems.rs:13]: Done

File: crates/hulk_replayer/src/frames.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (h3ndrk) [reply crates/hulk_replayer/src/frames.rs]: Up/down implemented

File: crates/hulk_replayer/src/frames.rs [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+        }
+        if keys.jump_forward_small {
+            *self.position += RelativeTime::new(1.0);
+        }
+        if keys.step_backward {
+            *self.position -= RelativeTime::new(0.01);
+        }
+        if keys.step_forward {
+            *self.position += RelativeTime::new(0.01);
+        }
```
[the remark this answers is not in the document]
Comment (h3ndrk) [reply crates/hulk_replayer/src/frames.rs:108]: Next iteration as discussed?

File: crates/hulk_replayer/src/ticks.rs [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+    fn show_position_text(
+        &self,
+        painter: &Painter,
+        font: FontId,
+        position: AbsoluteScreen,
+    ) -> Rect {
+        let clip_rect = painter.clip_rect();
+        let text =
+            absolute_time_to_date_time_string(self.position.map_to_absolute_time(self.frame_range));
```
[the remark this answers is not in the document]
Comment (h3ndrk) [reply crates/hulk_replayer/src/ticks.rs]: Changed to time strings only

**Review by knoellle — APPROVED** [review]
LGTM

**Pull request merged**